pyccel/parser/semantic.py

```diff
@@ -2574,6 +2574,8 @@ def _visit_Assign(self, expr, **settings):
                         bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                         severity='error')
                     return None
+            if lhs.is_temp and rhs.dtype is NativeVoid():
```

Comment:
tests are failing on this condition.

Reply:
Noticed that, on it.

Reply:
I think i fixed it in [42c9942](https://github.com/pyccel/pyccel/pull/1235/commits/42c994203c1136fb2642cb9972c317d421389383).